airflow-core/tests/unit/api_fastapi/common/test_exceptions.py

```diff
@@ -282,3 +295,91 @@ def test_handle_multiple_columns_unique_constraint_error(
 
             assert response_detail == expected_detail
             assert "INSERT INTO dag_run" in actual_statement
+        assert exeinfo_response_error.value.detail == expected_exception.detail
+
+
+class TestDagErrorHandler:
+    dag_error_handler = DagErrorHandler()
+
+    def test_handle_deserialization_error_with_value_error(self):
+        error_message = "Missing Dag ID in serialized Dag"
+        deserialization_error = DeserializationError("test_dag_id")
+
+        value_error = ValueError(error_message)
+        deserialization_error.__cause__ = value_error
+
+        with pytest.raises(
+            HTTPException,
+            match=re.escape(f"An error occurred while trying to deserialize Dag: {deserialization_error}"),
+        ):
+            self.dag_error_handler.exception_handler(None, deserialization_error)
+
+    def test_handle_deserialization_error_with_runtime_error(self):
+        error_message = "Error during Dag serialization process"
+        deserialization_error = DeserializationError("test_dag_id")
+
+        runtime_error = RuntimeError(error_message)
+        deserialization_error.__cause__ = runtime_error
+
+        expected_exception = HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An error occurred while trying to deserialize Dag: {deserialization_error}",
+        )
+
+        with pytest.raises(HTTPException, match=re.escape(expected_exception.detail)):
+            self.dag_error_handler.exception_handler(None, deserialization_error)
+
+    def test_handle_deserialization_error_with_key_error(self):
+        key = "required_field"
+        deserialization_error = DeserializationError("test_dag_id")
+
+        key_error = KeyError(key)
+        deserialization_error.__cause__ = key_error
+
+        expected_exception = HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An error occurred while trying to deserialize Dag: {deserialization_error}",
+        )
+        with pytest.raises(HTTPException, match=re.escape(expected_exception.detail)):
+            self.dag_error_handler.exception_handler(None, deserialization_error)
+
+    def test_handle_real_dag_deserialization_error(self):
```

Comment:
```suggestion
    def test_handle_real_dag_deserialization_error(self, session: Session) -> None:
```

I think we can use `session` fixture instead of creating the session everything

Reply:
These test comments are non-blocking. I'll merge and create another PR for refactoring